### PR TITLE
[Tests-only] Run tests with openIdConnect in CI

### LINKED ISSUE #3191: run oC 10 tests with openIDconnect
the future of auth is openidconnect, so we should test it

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Run acceptance tests with openIdConnect and ocis-phenix.

## Related Issue

- Fixes #3191

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -959,6 +982,79 @@ def cloneOauth():
...
+		'detach': True,
+		'environment' : {
+			'GLAUTH_BACKEND_DATASTORE': 'owncloud',
+			'LDAP_BASEDN': 'dc=example,dc=com',
+			'LDAP_BINDDN': 'dc=example,dc=com',
+			'GLAUTH_BACKEND_SERVERS': 'http://owncloud',
+		},
+		'commands': [
+			'cd /var/www/owncloud',
+			'./ocis-konnectd server'
```
Comment (phil-davis) [inline .drone.star]: Is this line correct?

File: .drone.star [hunk 2]
```diff
@@ -1292,7 +1386,6 @@ def redisService():
 		'name': 'redis',
 		'image': 'webhippie/redis',
 		'pull': 'always',
-		'detach': True,
```
Comment (kiranparajuli589) [inline .drone.star]: what is this `detach` for:question:

Comment (dpakach) [reply .drone.star]: We are now running redis as a service not step so we can remove that.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIopenIdLogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9200/

<p>















































































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIopenIdLogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9202/

<p>















































































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIopenIdLogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9203/

<p>















































































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIopenIdLogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9214/

<p>















































































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIopenIdLogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9216/

<p>















































































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIopenIdLogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9224/

<p>















































































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIopenIdLogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9225/

<p>






</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIopenIdLogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9230/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIopenIdLogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9231/

<p>






</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIopenIdLogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9258/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIopenIdLogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9259/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIopenIdLogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9263/

<p>





</p></details>

**Comment by dpakach** [comment]
encountering some strange errors while running tests with openIdConnect login.
1. When we run tests that delete multiple files at once in a loop, then the files will be locked immediately after deleting one files. 
2. When we upload a file overwriting an existing file immediately after logging in then the file is shown as locked. This fails intermittently.

These might be because of the problem in the server configuration, which I am investigating currently. I will report the issue and skip those tests if that is the case.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Upload</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9269/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Login</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9270/

<p>











</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Account</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/9270/

<p>





</p></details>

File: .drone.star [hunk 3]
```diff
@@ -1035,19 +1098,20 @@ def buildOcisPhoenix():
...
-			'PHOENIX_WEB_CONFIG': '/srv/config/drone/ocis-config.json',
-			'PHOENIX_ASSET_PATH': '/var/www/owncloud/phoenix/dist'
+			'PHOENIX_WEB_CONFIG': '/srv/config/drone/config.json' if glauth else '/srv/config/drone/ocis-config.json',
+			'PHOENIX_ASSET_PATH': '/var/www/owncloud/phoenix/dist',
+			'PHOENIX_OIDC_CLIENT_ID': 'phoenix'
 		},
 		'commands': [
 			'cd /var/www/owncloud',
-			'./ocis-phoenix server',
+			'./ocis-phoenix --log-level debug server',
```
Comment (individual-it) [inline .drone.star:1114]: do we want to keep that high log-level?

Comment (dpakach) [reply .drone.star:1114]: Yes, I think we should run all the ocis services in debug log level

**Comment by dpakach** [comment]
> does the description in the Readme, about how to run the test still apply?

Yes that still applies. The only thing different is that we run the tests now in CI with openId connect login. If we want to do that locally we can just set `OPENID_LOGIN=true`.

**Pull request merged**